assets/app/scripts/controllers/overview.js

```diff
@@ -384,7 +416,114 @@ angular.module('openshiftConsole')
       });
     });
 
+    /*
+     * The topology graph wants unique identifiers.
+     *
+     * One of the warts on the kubernetes/openshift API is that there
+     * is no unique global identifier per resource. We construct them here from
+     * Kind/metadata.name ... which is good enough in a single namespace.
+     */
+    function makeId(name, kind) {
```

Comment:
Please open a Kubernetes issue for that if one is not there already.

Sent from my iPhone

> On Jun 10, 2015, at 4:47 PM, Stef Walter <notifications@github.com> wrote:
> 
> In assets/app/scripts/controllers/overview.js:
> 
> > @@ -384,7 +416,114 @@ angular.module('openshiftConsole')
> >        });
> >      });
> >  
> > +    /*
> > +     * The topology graph wants unique identifiers.
> > +     *
> > +     * One of the warts on the kubernetes/openshift API is that there
> > +     * is no unique global identifier per resource. We construct them here from
> > +     * Kind/metadata.name ... which is good enough in a single namespace.
> > +     */
> > +    function makeId(name, kind) {
> The problem is that even if in theory a Deployment is a ReplicationController ... you still can't use a metadata.uid to refer to loaded resource objects (ie: loaded items) uniquely.
> 
> Even if you then assume that you will never use a ReplicationController and a Deployment resource objects together in a scenario where you would need unique identifiers ... you run straight into the second problem:
> 
> There are objects in kubernetes that do not have metadata.uid set. For example the 'kubernetes' and 'kubernetes-ro' services in the 'default' namespace.
> 
> —
> Reply to this email directly or view it on GitHub.
>